tests/readonlytest/readonly_test.go

```diff
@@ -130,41 +134,43 @@ func (s *TestReadOnlySuit) TestRestrictionWithConnectionPool(c *C) {
 
 	timer := time.NewTimer(10 * time.Second)
 	_, err = s.db.Exec("set global tidb_restricted_read_only=1")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	select {
 	case <-timer.C:
-		c.Fail()
+		require.Fail(t, "failed")
 	case success := <-done:
-		c.Assert(success, IsTrue)
+		require.True(t, success)
 	}
 }
 
-func (s *TestReadOnlySuit) TestReplicationWriter(c *C) {
+func TestReplicationWriter(t *testing.T) {
+	s, clean := createReadOnlySuite(t)
+	defer clean()
 	_, err := s.db.Exec("set global tidb_restricted_read_only=0")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	_, err = s.db.Exec("drop table if exists t")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	_, err = s.db.Exec("create table t (a int)")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	time.Sleep(1)
-	checkVariable(c, s.udb, false)
+	checkVariable(t, s.udb, false)
 
 	conn, err := s.rdb.Conn(context.Background())
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	go func(conn *sql.Conn) {
 		ticker := time.NewTicker(50 * time.Millisecond)
 		for {
-			t := <-ticker.C
-			_, err := conn.ExecContext(context.Background(), fmt.Sprintf("insert into t values (%d)", t.Nanosecond()))
-			c.Assert(err, IsNil)
+			t1 := <-ticker.C
+			_, err := conn.ExecContext(context.Background(), fmt.Sprintf("insert into t values (%d)", t1.Nanosecond()))
+			require.NoError(t, err)
 		}
 	}(conn)
 	time.Sleep(1 * time.Second)
 	timer := time.NewTimer(3 * time.Second)
 	_, err = s.db.Exec("set global tidb_restricted_read_only=1")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	// SUPER user can't write
 	_, err = s.db.Exec("insert into t values (1)")
-	c.Assert(err.Error(), Equals, ReadOnlyErrMsg)
+	require.NoError(t, err)
```

Comment:
Please check if it is equal to `ReadOnlyErrMsg`. There is `Error(t TestingT, err error, msgAndArgs ...interface{})`. Do not change the logic of tests easily.